### ISSUE TITLE
Support getting versions from VCS at a specified directory

### ISSUE THREAD

**Issue by vyasr** [mtkennerly/dunamai#69]
Although it's certainly not the best approach, it's not uncommon during development for packages to be made importable by simply adding their directory to the PYTHONPATH rather than installing them. Despite the advantages of modern editable installs I expect that the practice of modifying PYTHONPATH will persist. It seems that all of dunamai's logic for detecting versions (the `Version.from_*` methods) assumes that the current directory is the root, which is incompatible with packages added via PYTHONPATH. This could be overcome by using `dunamai.get_version` with a custom `third_choice` function that changes directories globally via `os.chdir`, but that's a fairly intrusive change to make in something like a package's `__init__.py` file where you might normally include dunamai code for version inference. Would it be possible to support a `root` or `path` kwarg for the `from_*` methods to indicate where to look for a package's git data and then use `os.path.join` internally to manage that without changing directories?

MRE:
```
(main) vyasr-dt% mkdir pkg
(main) vyasr-dt% cd pkg
(main) vyasr-dt% git init
Initialized empty Git repository in /home/vyasr/local/testing/dunamai_pythonpath/pkg/.git/
(main) vyasr-dt% echo "import dunamai" >> __init__.py
(main) vyasr-dt% echo "dunamai.get_version('pkg', third_choice=dunamai.Version.from_git)" >> __init__.py
(main) vyasr-dt% cd ..                          
(main) vyasr-dt% PYTHONPATH=${PWD}:${PYTHONPATH}
(main) vyasr-dt% cd /
(main) vyasr-dt% python -c "import pkg"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/vyasr/local/testing/dunamai_pythonpath/pkg/__init__.py", line 2, in <module>
    dunamai.get_version('pkg', third_choice=dunamai.Version.from_git)
  File "/home/vyasr/miniconda3/envs/main/lib/python3.8/site-packages/dunamai/__init__.py", line 2011, in get_version
    third_ver = third_choice()
  File "/home/vyasr/miniconda3/envs/main/lib/python3.8/site-packages/dunamai/__init__.py", line 1060, in from_git
    _detect_vcs(vcs)
  File "/home/vyasr/miniconda3/envs/main/lib/python3.8/site-packages/dunamai/__init__.py", line 355, in _detect_vcs
    raise RuntimeError(
RuntimeError: This does not appear to be a Git project
```

**Comment by mtkennerly** [comment]
> Would it be possible to support a `root` or `path` kwarg for the `from_*` methods

I think that's a good idea :) I'm open to adding it.

**Comment by mtkennerly** [comment]
This is now included in [v1.19.0](https://github.com/mtkennerly/dunamai/releases/tag/v1.19.0).
